ros2bag/ros2bag/verb/record.py

```diff
@@ -211,83 +211,63 @@ def _check_necessary_argument(self, args):
         # --topic-types or --regex must be used
         if not (args.all or args.all_topics or args.all_services or
-           args.services or (args.topics and len(args.topics) > 0) or
-           (args.topic_types and len(args.topic_types) > 0) or args.regex):
+                args.services or (args.topics and len(args.topics) > 0) or
+                (args.topic_types and len(args.topic_types) > 0) or args.regex):
             return False
         return True
 
-    def main(self, *, args):  # noqa: D102
-
+    def validate_parsed_arguments(self, args, uri) -> str:
         if args.topics_positional:
-            print('Warning! Positional "topics" argument deprecated. '
-                  'Please use optional "--topics" argument instead.')
+            print(print_warn('Positional "topics" argument deprecated. '
+                             'Please use optional "--topics" argument instead.'))
             args.topics = args.topics_positional
 
         if not self._check_necessary_argument(args):
-            return print_error('Need to specify one option out of --all, --all-topics, '
-                               '--all-services, --services, --topics, --topic-types and --regex')
-
-        # Only one option out of --all, --all-services --services or --regex can be used
-        if (args.all and args.all_services) or \
-           ((args.all or args.all_services) and args.regex) or \
-           ((args.all or args.all_services or args.regex) and args.services):
-            return print_error('Must specify only one option out of --all, --all-services, '
-                               '--services or --regex')
-
-        # Only one option out of --all, --all-topics, --topics, --topic-types or --regex can
-        # be used
-        if (args.all and args.all_topics) or \
-           ((args.all or args.all_topics) and args.regex) or \
-           ((args.all or args.all_topics or args.regex) and args.topics) or \
-           ((args.all or args.all_topics or args.regex or args.topics) and args.topic_types):
-            return print_error('Must specify only one option out of --all, --all-topics, '
-                               '--topics, --topic-types or --regex')
-
-        if (args.exclude_regex and
-           not (args.regex or args.all or args.all_topics or args.all_services)):
-            return print_error('--exclude-regex argument requires either --all, '
-                               '--all-topics, --all-services or --regex')
-
-        if args.exclude_topics and not (args.regex or args.all or args.all_topics):
-            return print_error('--exclude-topics argument requires either --all, --all-topics '
-                               'or --regex')
-
-        if args.exclude_topic_types and not (args.regex or args.all or args.all_topics):
-            return print_error('--exclude-topic-types argument requires either --all, '
-                               '--all-topics or --regex')
```

Comment:
The comments are the same as above.  
Besides, `args.topics` should be added.

Reply:
No. this check is not necessary we already do check for must-have inclusion arguments beforehand in the
https://github.com/ros2/rosbag2/pull/1633/files#diff-69df89597d099d6eeeb32ecc480c22116dc6c97f32556ef9636eee901f62452cR210-R215

Reply:
Fixed in the https://github.com/ros2/rosbag2/pull/1633/commits/9eb6641595aa55ea8c28c88b7f2a777bb21d86dd